lib/spi/types.ts

```diff
@@ -123,7 +123,7 @@ export class ResourceContext {
 
     private readonly resources: Map<string, IConstruct> = new Map();
 
-    constructor(public readonly scope: cdk.Stack, public readonly blueprintProps: EksBlueprintProps) { }
+    constructor(public readonly scope: any, public readonly blueprintProps: EksBlueprintProps) { }
```

Comment:
Why scope is any? Let's make it `Construct`.

Reply:
Yes, Contruct was my first choice. But I ran into issues while doing that. So didn't want to refactor/change too much of current code. So, I used any type to be generic.

Issues because of Construct:

lib/resource-providers/hosted-zone.ts:92:41 - error TS2339: Property 'formatArn' does not exist on type 'Construct'.

92         const delegationRoleArn = stack.formatArn({
                                           ~~~~~~~~~

lib/resource-providers/iam.ts:60:43 - error TS2339: Property 'account' does not exist on type 'Construct'.

60             `arn:aws:iam::${context.scope.account}:oidc-provider/${this.url.substring(httpsPrefix.length)}`

Reply:
@bugathagit I am glad compiler helps us catch these issues. So for the places when stack is needed, let's use Stack.of(context.scope), it is a minor change that will go into the hosted zone 
`const stack = context.scope; ` becomes `const stack = Stack.of(context.scope);`.

Reply:
Thank you!! Updated the same in commit `1fe12f9a014c0c90be7cf91e0b2bb000f71f5c16`